explainer-beforematch.md

```diff
@@ -191,6 +191,86 @@ In general, we believe that the `beforematch` event does not pose a privacy prob
 However, the privacy aspect of the event should be discussed in more detail via
 formal security reviews.
 
+### Responses to DOM and style changes in the `beforematch` event handler
+
+The scrolling behavior exhibited by find-in-page, ScrollToTextFragment, and
+ElementFragment after style and DOM changes are made by `beforematch` event
+handlers is not well specified yet and is being discussed here: 
+https://github.com/WICG/display-locking/issues/150
+
+Here is a list of `beforematch` handler use cases that could affect scrolling:
+1. **Removed from dom**: The `beforematch` event handler removes the
+target element from the dom.
+2. **Reparented in dom**: The `beforematch` event handler removes the target
+element from the dom, and re-adds it further down the tree such that the
+location to scroll to is different.
+3. **display: none**: The `beforematch` event handler adds the style
+`display: none` to the target element.
+4. **visibility: hidden**: The `beforematch` event handler adds the style
+`visibility: hidden` to the target element.
+5. **subtree-visibility: hidden-matchable -> visible**: The target element has
+the style `subtree-visibility: hidden-matchable` before the `beforematch` event
+is fired on the target element, and the `beforematch` event handler changes the
+style value from `hidden-matchable` to `visible`.
+
+
+Here is the current scrolling behavior in chromium for each of those cases in
+ScrollToTextFragment, and ElementFragment, and find-in-page:
+
+#### ScrollToTextFragment
+1. **Removed from dom**: ScrollToTextFragment will not scroll to the removed
+element. If there is a second match which is still in the dom after
+beforematch, it will be scrolled to.
+2. **Reparented in dom**: ScrollToTextFragment will scroll to the new
+location of the target. If there is a second match which was not reparented,
+ScrollToTextFragment will scroll to whichever comes first after the beforematch
```

Comment:
Comes first in what order? From the top of the page? Or from where the initial match was?

Reply:
From the top of the page. I'll clarify here.